internal/prober/prober_test.go

```diff
@@ -33,192 +42,159 @@ import (
+	mockShootClientCreatorError error
+	mockDiscoveryError          error
+	mockLeaseListError          error
+	mockScaleUpError            error
+	mockScaleDownError          error
+	minMockScaleUpCount         int
+	maxMockScaleUpCount         int
+	minMockScaleDownCount       int
+	maxMockScaleDownCount       int
 }
 
-func TestInternalProbeErrorCount(t *testing.T) {
-	table := []probeStatusEntry{
-		{"Success Count is incremented", nil, 1, 0, 1, 0},
-		{"Unignorable error is returned by pingKubeApiServer", errNotIgnorable, 0, 1, 0, 0},
-		{"Forbidden request error is returned by pingKubeApiServer", apierrors.NewForbidden(schema.GroupResource{}, "test", errors.New("forbidden")), 0, 0, 0, 0},
-		{"Unauthorized request error is returned by pingKubeApiServer", apierrors.NewUnauthorized("unauthorized"), 0, 0, 0, 0},
-		{"Throttling error is returned by pingKubeApiServer", apierrors.NewTooManyRequests("Too many requests", 10), 0, 0, 0, 0},
+func TestAPIServerProbeFailure(t *testing.T) {
+	table := []probeSimulatorEntry{
+		{"Unknown error is returned by api server", nil, nil, unknownError, nil, nil, nil, 0, 0, 0, 0},
+		{"Forbidden request error is returned by api server", nil, nil, apierrors.NewForbidden(schema.GroupResource{}, "test", errors.New("forbidden")), nil, nil, nil, 0, 0, 0, 0},
+		{"Unauthorized request error is returned by api server", nil, nil, apierrors.NewUnauthorized("unauthorized"), nil, nil, nil, 0, 0, 0, 0},
+		{"Throttling error is returned by api server", nil, nil, apierrors.NewTooManyRequests("Too many requests", 10), nil, nil, nil, 0, 0, 0, 0},
 	}
 
 	for _, entry := range table {
 		t.Run(entry.name, func(t *testing.T) {
-			setupProberTest(t)
-			config = createConfig(1, 1, metav1.Duration{Duration: 4 * time.Millisecond}, metav1.Duration{Duration: time.Microsecond}, 0.2)
-
-			msc.EXPECT().CreateClient(gomock.Any(), proberTestLogger, gomock.Any(), gomock.Any(), gomock.Any()).Return(mki, nil).AnyTimes()
-			mki.EXPECT().Discovery().Return(mdi).AnyTimes()
-			mdi.EXPECT().ServerVersion().Return(nil, entry.err).AnyTimes()
-			mds.EXPECT().ScaleUp(gomock.Any()).Return(nil).AnyTimes()
-
-			runProberAndCheckStatus(t, 12*time.Millisecond, entry)
+			createMockInterfaces(t)
+			initializeMockInterfaces(entry)
+			config := createConfig(testProbeInterval, metav1.Duration{Duration: time.Microsecond}, metav1.Duration{Duration: 40 * time.Second}, 0.2)
+			createAndRunProber(t, testProbeInterval.Duration, config, entry)
 		})
 	}
 }
 
-func TestHealthyProbesShouldRunScaleUp(t *testing.T) {
-	table := []probeStatusEntry{
-		{"Scale Up Succeeds", nil, 1, 0, 1, 0},
-		{"Scale Up Fails", errors.New("scale Up failed"), 1, 0, 1, 0},
-	}
-
-	for _, probeStatusEntry := range table {
-		t.Run(probeStatusEntry.name, func(t *testing.T) {
-			setupProberTest(t)
-			config = createConfig(1, 1, metav1.Duration{Duration: 4 * time.Millisecond}, metav1.Duration{Duration: time.Microsecond}, 0.2)
+func TestSuccessfulProbesShouldRunScaleUp(t *testing.T) {
```

Comment:
since you are using files, therefore you are limited to only 2 leases. Just create a `NodeLeaseCreator` which takes in a bunch of parameters allowing to to construct `coordinationv1.Lease`objects programmatically.